src/Microsoft.DotNet.Core.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Extensions.DependencyModel;
+using Newtonsoft.Json;
+using NuGet.Frameworks;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Core.Build.Tasks
+{
+    /// <summary>
+    /// Generates the $(project).deps.json file.
+    /// </summary>
+    public class GenerateDepsFile : Task
+    {
+        [Required]
+        public string LockFilePath { get; set; }
+
+        [Required]
+        public string DepsFilePath { get; set; }
+
+        [Required]
+        public string TargetFramework { get; set; }
+
+        public string RuntimeIdentifier { get; set; }
+
+        [Required]
+        public string AssemblyName { get; set; }
+
+        [Required]
+        public string AssemblyVersion { get; set; }
+
+        public override bool Execute()
+        {
+            LockFile lockFile = LockFileCache.Instance.GetLockFile(LockFilePath);
```

Comment:
I think that can be done when/if this class is unit tested.  The current approach is that this Task is so simple and has no logic in it that it doesn't need to be unit tested.  Instead, the unit tests are targeted at the class with the logic in it: DependencyContextBuilder.